### PR TITLE
Settings object for menuitemradio mode

### PR DESCRIPTION
**Pull request by Heydon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,6 +98,17 @@ const exampleButton = document.querySelector('[data-opens-menu]')
 const exampleMenuButton = new MenuButton(exampleButton)
 ```
 
+### `menuitemradio` example
```
Comment (KittyGiraudel) [inline README.md:101]: I would make this a bit more explicit. The fact that it uses `menuitemradio` is not really the first thing the user wants to know.

File: README.md [hunk 2]
```diff
@@ -94,6 +98,17 @@ const exampleButton = document.querySelector('[data-opens-menu]')
...
+
+Sometimes you'd like to persist the selected menu item, using a checked state. In this case, the script should implement `role="menuitemradio"` in place of `role="menuitem"`, and apply `aria-checked="true"` to the most recently chosen item. On initialization, supply `radios: true` to the constructor call:
+
+```js
+// Make it a menu button with menuitemradio buttons
+const exampleMenuButton = new MenuButton(exampleButton, {radios: true})
```
Comment (KittyGiraudel) [inline README.md:107]: Missing spaces inside braces. ;)

File: README.md [hunk 3]
```diff
@@ -94,6 +98,17 @@ const exampleButton = document.querySelector('[data-opens-menu]')
...
+```
+
+The basic CSS (see above) prefixes the checked item with a check mark. This declaration can be removed safely and replaced with a different form of indication.
```
Comment (KittyGiraudel) [inline README.md]: That’s nice and clear.

File: examples/menuitemradio.html [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+       <button>Very Tricky</button>
+       <button>Incredibly Hard</button>
+       <button>Impossible</button>
+     </div>
+    </div>
+    <script src="../inclusive-menu-button.js"></script>
+    <script>
+      // Instantiation
+      var exampleButton = document.querySelector('[data-inclusive-menu-opens]')
+      var exampleMenuButton = new MenuButton(exampleButton, {radios: true})
```
Comment (KittyGiraudel) [inline examples/menuitemradio.html:59]: Missing spaces inside braces. ;)

File: examples/menuitemradio.html [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+      // Listen to choose event
+      exampleMenuButton.on('choose', function(choice) {
```
Comment (KittyGiraudel) [inline examples/menuitemradio.html:62]: Missing space before function parentheses. ;)

File: inclusive-menu-button.js [hunk 6]
```diff
@@ -4,7 +4,26 @@
   'use strict'
 
   // Constructor
-  function MenuButton (button) {
+  function MenuButton (button, userSettings) {
+    // Error if the userSettings argument isn't an object
+    if (userSettings !== undefined && typeof userSettings !== 'object') {
+      throw new Error('MenuButton\'s second argument is expected as an object or undefined')
```
Comment (KittyGiraudel) [inline inclusive-menu-button.js:10]: You could (should?) use a curly quote to avoid handing to escape.

File: inclusive-menu-button.js [hunk 7]
```diff
@@ -4,7 +4,26 @@
   'use strict'
 
   // Constructor
-  function MenuButton (button) {
+  function MenuButton (button, userSettings) {
+    // Error if the userSettings argument isn't an object
+    if (userSettings !== undefined && typeof userSettings !== 'object') {
```
Comment (KittyGiraudel) [inline inclusive-menu-button.js:9]: Use `typeof userSettings !== 'undefined` instead.

File: inclusive-menu-button.js [hunk 8]
```diff
@@ -4,7 +4,26 @@
   'use strict'
 
   // Constructor
-  function MenuButton (button) {
+  function MenuButton (button, userSettings) {
```
Comment (KittyGiraudel) [inline inclusive-menu-button.js:7]: I’d just name it `settings` or `options`.

File: inclusive-menu-button.js (hunk 7 above)
Comment (KittyGiraudel) [inline inclusive-menu-button.js:9]: Side note, `null` will go through here (as it is technically an object).

File: inclusive-menu-button.js [hunk 9]
```diff
@@ -4,7 +4,26 @@
...
+    if (userSettings !== undefined && typeof userSettings !== 'object') {
+      throw new Error('MenuButton\'s second argument is expected as an object or undefined')
+    }
+
+    // The default settings
+    this.settings = {
+      radios: false
+    }
+
+    // Overwrite defaults where they are provided in userSettings
```
Comment (KittyGiraudel) [inline inclusive-menu-button.js:18]: If [support](http://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign) is okay, you could (should?) do:

```js
this.settings = Object.assign({
  radios: false
}, userSettings)
```

File: inclusive-menu-button.js [hunk 10]
```diff
@@ -4,7 +4,26 @@
...
-  function MenuButton (button) {
+  function MenuButton (button, userSettings) {
+    // Error if the userSettings argument isn't an object
+    if (userSettings !== undefined && typeof userSettings !== 'object') {
+      throw new Error('MenuButton\'s second argument is expected as an object or undefined')
+    }
+
+    // The default settings
+    this.settings = {
+      radios: false
```
Comment (KittyGiraudel) [inline inclusive-menu-button.js:15]: I’m not too sure about the naming here to be faire. Maybe something like `checkable: <bool>` would be more obvious?

**Review by KittyGiraudel — APPROVED** [review]

**Comment by Heydon** [comment]
Thank you, Hugo. I have moved on to allow `menuitemcheckbox` as well. I will close this pull request and merge that one instead, with your advice taken.

BTW, `Object.assign` is not supported in Internet Explorer at all :-(

**Pull request closed without merging**